src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -150,13 +159,53 @@ public PersistentDataStore() {
     return messages;
   }
 
+  /**
+   * Loads all Mention objects from the Datastore service and returns them in a List, sorted in
+   * ascending order by creation time.
+   *
+   * @throws PersistentDataStoreException if an error was detected during the load from the
+   *     Datastore service
+   */
+  public List<Mention> loadMentions() throws PersistentDataStoreException {
+
+    List<Mention> mentions = new ArrayList<>();
+
+    // Retrieve all messages from the datastore.
+    Query query = new Query("chat-mentions");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      try {
+        List<String> dataStoreMessageIds = new ArrayList((Collection<String>) entity.getProperty("uuid_list")); 
+        Set<UUID> messageIds = new HashSet<UUID>();
+        String mentionedUser = (String) entity.getProperty("mentioned_user");
+        
+        for (String uuid: dataStoreMessageIds) { 
+          messageIds.add(UUID.fromString(uuid));
+        }
+        Mention mention = new Mention(messageIds, mentionedUser);
```

Comment:
The constructor for this uses a Set, I would constuct messageIds as a Set instead of a List

`Set<String> datastoreMessageIds = new HashSet<>((Collection<String>) entity.getProperty("uuid_list"));`

You also never fill messageIds with anything